### PR TITLE
Add things the frontend needs

### PR DESCRIPTION
**Pull request by gcampo88**
- [X] allow PUT requests
- [x] add updatedAt to campaign API show response
- [x] add fields to campaigns:
  - originalCopy vs. editedCopy
  - notes
  - mustSay
  - dueDate
  - approved (only if sponsor requires approval) 
- [x] update the google app script to add a campaign to reflect new fields

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,4 +31,4 @@ erl_crash.dump
 /deps_docker_compose
 
 # seeds
-*seeds.exs
+/priv/repo/seeds.exs
```
Comment (cavis) [inline .gitignore]: Why ignore this now?  (I had assumed you wanted these dev seeds checked in).

File: priv/repo/migrations/20171213153434_add_fields_to_campaigns.exs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  use Ecto.Migration
+
+  def change do
+    rename table(:campaigns), :copy, to: :original_copy
+
+    alter table(:campaigns) do
+      add :edited_copy, :text
+      add :must_say, :text
+      add :due_date, :date
+      add :notes, :string
```
Comment (cavis) [inline priv/repo/migrations/20171213153434_add_fields_to_campaigns.exs]: Why not `:text`?  Seems like you might end up with > 255 chars of notes.

File: web/models/podcast.ex [hunk 3]
```diff
@@ -1,4 +1,4 @@
-defmodule Jingle.API.Podcast do
+defmodule Jingle.Podcast do
```
Comment (cavis) [inline web/models/podcast.ex:1]: :+yes:

File: web/models/campaign.ex [hunk 4]
```diff
@@ -18,8 +23,9 @@ defmodule Jingle.Campaign do
   """
   def changeset(struct, params \\ %{}) do
     struct
-    |> cast(params, [:sponsor_id, :podcast_id, :start_date, :end_date, :copy, :zone])
-    |> validate_required([:sponsor_id, :podcast_id, :start_date, :end_date, :copy, :zone])
+    |> cast(params, [:sponsor_id, :podcast_id, :start_date, :end_date, :due_date, :original_copy, :edited_copy, :must_say, :notes, :approved, :zone])
+    |> validate_required([:start_date, :end_date, :original_copy, :zone])
```
Comment (cavis) [inline web/models/campaign.ex:27]: Sponsor/podcast relations no longer required?

File: lib/jingle/endpoint.ex [hunk 5]
```diff
@@ -42,7 +42,7 @@ defmodule Jingle.Endpoint do
   plug Corsica,
     origins: ~r/.*\.prx\.(?:org|dev|tech|docker)$/,
     allow_headers: ~w(Authorization),
-    allow_methods: ~w(HEAD GET),
+    allow_methods: ~w(HEAD GET PUT),
```
Comment (cavis) [inline lib/jingle/endpoint.ex]: Remind me ... how has this not already caused problems?

(Also, does this need to include POST/DELETE as well for frontend CRUD to work?)

**Review by cavis — CHANGES_REQUESTED** [review]
Looks pretty good to me.  Mostly just a bunch of help-me-be-understandful questions.

File: .gitignore (hunk 1 above)
Comment (gcampo88) [reply .gitignore]: no particular reason! hadn't realized i'd checked them in before. but they don't hurt. will remove this!

File: lib/jingle/endpoint.ex (hunk 5 above)
Comment (gcampo88) [reply lib/jingle/endpoint.ex]: frontend hadn't been doing anything but GET yet! and -- yes, Creatives will be created + destroyed through frontend. Might as well add that now! will add.

File: priv/repo/migrations/20171213153434_add_fields_to_campaigns.exs (hunk 2 above)
Comment (gcampo88) [reply priv/repo/migrations/20171213153434_add_fields_to_campaigns.exs]: seems reasonable! will change

File: web/models/campaign.ex (hunk 4 above)
Comment (gcampo88) [reply web/models/campaign.ex:27]: next lines take care of!:    
```elixir
    |> assoc_constraint(:sponsor)
    |> assoc_constraint(:podcast)
```

**Comment by gcampo88** [comment]
back to you sir @cavis

File: web/models/campaign.ex (hunk 4 above)
Comment (cavis) [reply web/models/campaign.ex:27]: ah, gotcha.  good call!

**Review by cavis — APPROVED** [review]
:plus1:

**Pull request merged**